sky/backends/backend_utils.py

```diff
@@ -1609,17 +1609,19 @@ def _query_status_gcp(
     # TODO(wei-lin): handle multi-node cases.
     if use_tpu_vm and len(status_list) == 0:
         logger.debug(f'Terminating preempted TPU VM cluster {cluster}')
-        # Do not use CloudVMRayBackend.teardown_no_lock, as that will
+        backend = backends.CloudVmRayBackend()
+        handle = global_user_state.get_handle_from_cluster_name(cluster)
+        # Do not use refresh cluster status during teardown, as that will
         # cause inifinite recursion by calling cluster status refresh
         # again.
-        terminate_cmd = tpu_utils.terminate_tpu_vm_cluster_cmd(cluster, zone)
-        returncode, stdout, stderr = log_lib.run_with_log(terminate_cmd,
-                                                          os.devnull,
-                                                          shell=True,
-                                                          stream_logs=False,
-                                                          require_outputs=True)
-        subprocess_utils.handle_returncode(returncode, terminate_cmd,
-                                           stdout + stderr)
+        # The only caller of this function, `_get_cluster_status_via_cloud_cli`,
```

Comment:
```suggestion
        # The caller of this function, `_update_cluster_status_no_lock() -> _get_cluster_status_via_cloud_cli()`,
```
Was a typo before. Post teardown cleanup is in `_update_cluster_status_no_lock`.